litmus/sched_plugin.c

```diff
@@ -253,11 +267,30 @@ int register_sched_plugin(struct sched_plugin* plugin)
 	list_add(&plugin->list, &sched_plugins);
 	raw_spin_unlock(&sched_plugins_lock);
 
-	return 0;
+out_reg:
+	return err;
+}
+EXPORT_SYMBOL(register_sched_plugin);
+
+int unregister_sched_plugin(struct sched_plugin* plugin)
+{
+	int unregister = 0;
+
+	if (strcmp(litmus->plugin_name, plugin->plugin_name) != 0)
+	{
+		unregister = 1;
+		raw_spin_lock(&sched_plugins_lock);
+		list_del(&plugin->list);
+		raw_spin_unlock(&sched_plugins_lock);
+	}
+	else
+		printk(KERN_ALERT "The currently active plugin can not be removed\n");
```

Comment:
So what happens if this case is triggered during a module_exit() callback? How can it prevent the unloading at this point?

Reply:
Answered at the pull request conversation